packages/react-instantsearch-dom/src/hooks/useAnswers.js

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useEffect, useMemo, useContext } from 'react';
+import { instantSearchContext } from 'react-instantsearch-core';
+import { createConcurrentSafePromise } from '../lib/createConcurrentSafePromise';
+import { debounce, debounceAsync } from '../lib/debounce';
+
+function hasReactHooks() {
+  // >= 16.8.0
+  const [major, minor] = React.version.split('.').map(Number);
+  return major >= 17 || (major === 16 && minor >= 8);
+}
+
+export default function useAnswers({
+  searchClient,
+  queryLanguages,
+  attributesForPrediction,
+  nbHits,
+  renderDebounceTime = 100,
+  searchDebounceTime = 100,
+  ...extraParameters
+}) {
+  if (!hasReactHooks()) {
+    throw new Error(
+      `\`Answers\` component and \`useAnswers\` hook require all React packages to be 16.8.0 or higher.`
+    );
+  }
+  const context = useContext(instantSearchContext);
+  const [query, setQuery] = useState();
+  const [index, setIndex] = useState();
+  const [isLoading, setIsLoading] = useState();
+  const [hits, setHits] = useState();
+  const runConcurrentSafePromise = useMemo(
+    () => createConcurrentSafePromise(),
+    []
+  );
+  const searchIndex = useMemo(() => searchClient.initIndex(index), [
+    searchClient,
+    index,
+  ]);
+  const debouncedSearch = useMemo(() => {
+    if (!searchIndex) {
+      // eslint-disable-next-line prefer-promise-reject-errors
+      return () => Promise.reject();
+    }
+    if (!searchIndex.findAnswers) {
+      throw new Error(
+        '`Answers` component and `useAnswers` hook require `algoliasearch` to be 4.8.0 or higher.'
+      );
+    }
```

Comment:
this error should be thrown at line 39 instead of in the search call itself I think